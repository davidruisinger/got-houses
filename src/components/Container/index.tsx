import styled from '../_utils/styledComponents'
import { MEDIA_BREAKPOINTS } from '../_utils/mediaQuery'

interface Props {
  textAlign?: 'left' | 'right' | 'center'
}

const Container = styled.div<Props>`
  width: 100%;
  max-width: ${MEDIA_BREAKPOINTS.lg}px;
  margin: 0 auto;
  padding: 1em;
  box-sizing: border-box;
  overflow: hidden;
  text-align: ${props => props.textAlign};
`

Container.defaultProps = {
  textAlign: 'left',
}

export default Container
