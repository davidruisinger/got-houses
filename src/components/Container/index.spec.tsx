import 'jest-styled-components'
import React from 'react'
import testRenderer from 'react-test-renderer'
import Container from './index'

describe('<Container />', () => {
  const tree = testRenderer.create(<Container />).toJSON()
  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
  it('has "text-align" set to "left"', () => {
    expect(tree).toHaveStyleRule('text-align', 'left')
  })
})

describe('<Container textAlign="right" />', () => {
  const tree = testRenderer.create(<Container textAlign="right" />).toJSON()
  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
  it('has "text-align" set to "left"', () => {
    expect(tree).toHaveStyleRule('text-align', 'right')
  })
})

describe('<Container textAlign="center" />', () => {
  const tree = testRenderer.create(<Container textAlign="center" />).toJSON()
  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
  it('has "text-align" set to "left"', () => {
    expect(tree).toHaveStyleRule('text-align', 'center')
  })
})
