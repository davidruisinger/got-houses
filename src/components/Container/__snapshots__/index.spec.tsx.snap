// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> matches the snapshot 1`] = `
.c0 {
  width: 100%;
  max-width: 992px;
  margin: 0 auto;
  padding: 1em;
  box-sizing: border-box;
  overflow: hidden;
  text-align: left;
}

<div
  className="c0"
/>
`;

exports[`<Container textAlign="center" /> matches the snapshot 1`] = `
.c0 {
  width: 100%;
  max-width: 992px;
  margin: 0 auto;
  padding: 1em;
  box-sizing: border-box;
  overflow: hidden;
  text-align: center;
}

<div
  className="c0"
/>
`;

exports[`<Container textAlign="right" /> matches the snapshot 1`] = `
.c0 {
  width: 100%;
  max-width: 992px;
  margin: 0 auto;
  padding: 1em;
  box-sizing: border-box;
  overflow: hidden;
  text-align: right;
}

<div
  className="c0"
/>
`;
